platform/api.templates/src/org/netbeans/api/templates/CreateFromTemplateHandler.java

```diff
@@ -59,4 +63,169 @@
             CreateDescriptor    desc
     ) throws IOException;
 
+    /**
+     * Replaces ${param} tokens in a String. The parameter reference has syntax "${" paramName [":" defaultValue ] "}".
+     * Parameters are replaced recursively. "${" can be escaped by \, so "\${" will not act as parameter reference start.
+     * @param expression string template
+     * @param parameters parameter values
+     * @return evaluated string.
+     */
+    static String mapParameters(String expression, Map<String, ?> parameters) {
```

Comment:
I was hoping this _in-house mapping_ can now be removed when we can use the Freemarker based one (with `'...'`) easily. But I can live with both variants. Up to you.